x-pack/plugins/reporting/server/export_types/csv_searchsource_immediate/execute_job.ts

```diff
@@ -32,9 +32,10 @@ export const runTaskFnFactory: RunTaskFnFactory<ImmediateExecuteFn> = function e
   const config = reporting.getConfig();
   const logger = parentLogger.clone([CSV_SEARCHSOURCE_IMMEDIATE_TYPE, 'execute-job']);
 
-  return async function runTask(jobId, immediateJobParams, context, req) {
+  return async function runTask(_jobId, immediateJobParams, context, req) {
     const job = {
       objectType: 'immediate-search',
+      version: reporting.getKibanaVersion(),
```

Comment:
why do we need to provide the version here ? from my understanding:
- version is provided from the client side (in the generate report url or when clicking the generate report in ui)
- server checks if version is provided and if not fallbacks to 7.14.0

Reply:
It was because the `job` object gets passed directly to `GenerateCSV`, since this is an "immediate download" export - not a queued job. which had the parameter typed as `JobParamsCSV` and therefore needed the `version`.

This can be simplified by adding `Omit<>` around the type in the parameter: 300ba45f6c4